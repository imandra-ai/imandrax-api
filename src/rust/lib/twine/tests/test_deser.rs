use anyhow::Result;
use insta::assert_debug_snapshot; // see: https://docs.rs/insta/latest/insta/
use std::fs;

#[test]
fn test_pubsub() -> Result<()> {
    let data = fs::read("tests/test_data/pubsub.twine")?;
    assert_debug_snapshot!(data.len());
    let d = twine::Decoder::new(&data)?;
    let bump = bumpalo::Bump::new();
    //println!("data:");
    //println!("{}", pretty_hex::pretty_hex(&data));
    let v = twine::value::get_value_from_entrypoint(&d, &bump)?;
    assert_debug_snapshot!(v);
    Ok(())
}

#[test]
fn test_typereg() -> Result<()> {
    let data = fs::read("tests/test_data/typereg.twine")?;
    assert_debug_snapshot!(data.len());
    let d = twine::Decoder::new(&data)?;
    let bump = bumpalo::Bump::new();
    //println!("data:");
    //println!("{}", pretty_hex::pretty_hex(&data));
    let v = twine::value::get_value_from_entrypoint(&d, &bump)?;
    assert_debug_snapshot!(v);
    Ok(())
}
