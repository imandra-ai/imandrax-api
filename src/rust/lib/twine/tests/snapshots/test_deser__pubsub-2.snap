---
source: tests/test_deser.rs
expression: v
---
Dict(
    [
        (
            Imm(
                String(
                    "PUBSUB",
                ),
            ),
            Dict(
                [
                    (
                        Imm(
                            String(
                                "summary",
                            ),
                        ),
                        Imm(
                            String(
                                "A container for Pub/Sub commands.",
                            ),
                        ),
                    ),
                    (
                        Imm(
                            String(
                                "complexity",
                            ),
                        ),
                        Imm(
                            String(
                                "Depends on subcommand.",
                            ),
                        ),
                    ),
                    (
                        Imm(
                            String(
                                "group",
                            ),
                        ),
                        Imm(
                            String(
                                "pubsub",
                            ),
                        ),
                    ),
                    (
                        Imm(
                            String(
                                "since",
                            ),
                        ),
                        Imm(
                            String(
                                "2.8.0",
                            ),
                        ),
                    ),
                    (
                        Imm(
                            String(
                                "arity",
                            ),
                        ),
                        Imm(
                            Int64(
                                -2,
                            ),
                        ),
                    ),
                ],
            ),
        ),
    ],
)
